Dockerfile.ubi

```diff
@@ -57,23 +57,30 @@ ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
 # install cuda and common dependencies
 RUN --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=cache,target=/root/.cache/uv \
     --mount=type=bind,source=requirements-common.txt,target=requirements-common.txt \
     --mount=type=bind,source=requirements-cuda.txt,target=requirements-cuda.txt \
-    pip install \
+    uv pip install \
         -r requirements-cuda.txt
 
+# force using the python venv's cuda runtime libraries
+ENV LD_LIBRARY_PATH="${VIRTUAL_ENV}/lib/python${PYTHON_VERSION}/site-packages/nvidia/cuda_nvrtc/lib:${LD_LIBRARY_PATH}"
+ENV LD_LIBRARY_PATH="${VIRTUAL_ENV}/lib/python${PYTHON_VERSION}/site-packages/nvidia/cuda_runtime/lib:${LD_LIBRARY_PATH}"
```

Comment:
It looks like that even hardcoding `LD_LIBRARY_PATH` doesn't change how `cmake` finds nvrtc:

```
#25 10.18 -- CUDART: /usr/local/cuda/lib64/libcudart.so
#25 10.18 -- CUDA Driver: /usr/local/cuda/lib64/stubs/libcuda.so
#25 10.18 -- NVRTC: /usr/local/cuda/lib64/libnvrtc.so
```

but, assuming that the versions match (:facepalm:), the built libraries might be happy with what they find.,